### DISCUSSION TITLE
Serde alias macro doesn't work.

### DISCUSSION THREAD

**Discussion by cbishopvelti** [SeaQL/sea-orm#1497] [Q&A]
## Description

When using #[serde(alias="xyz")]  from_json, the data doesn't get mapped into the ActiveModel. 

## Steps to Reproduce

```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
    pub struct Model {
      #[sea_orm(primary_key)]
      #[serde(skip_deserializing)]
      pub id: i32,

      #[serde(alias="aB")]
      a_b: String
    }

    #[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
    pub enum Relation {}

    impl ActiveModelBehavior for ActiveModel {}

    #[test]
    fn test_serde() {
      let test = json!({
        "aB": "ab"
      });

      let out = ActiveModel::from_json(test);

      println!("001 {:?}", out);

    }
```

### Expected Behavior

001 Ok(ActiveModel { id: NotSet, a_b: Set("ab") })

### Actual Behavior

001 Ok(ActiveModel { id: NotSet, a_b: NotSet })

### Reproduces How Often

Always

## Versions

```
├── sea-orm v0.11.0
│   ├── sea-orm-macros v0.11.0 (proc-macro)
│   ├── sea-query v0.28.3
│   │   ├── sea-query-derive v0.3.0 (proc-macro)
│   ├── sea-query-binder v0.3.0
│   │   ├── sea-query v0.28.3 (*)
│   ├── sea-strum v0.23.0
│   │   └── sea-strum_macros v0.23.0 (proc-macro)
```

**Comment by siketyan** [comment]
Deriving ActiveModel does not copy the attributes on fields excepting sea_orm attrs.

https://github.com/SeaQL/sea-orm/blob/2e1cfed8b5472c56be4f6fea03e11d34ad961456/sea-orm-macros/src/derives/active_model.rs#L53

**Comment by siketyan** [comment]
There is a limitation of Rust that procedural macros can not call another macros; you can declare another struct to be deserialised from JSON as a workaround.

**Comment by tyt2y3** [comment]
Well said, @siketyan . Thank you!

I think we can't do much more here about this issue, so I'd close it now.

**Comment by cbishopvelti** [comment]
Can't there be an attribute on the sea_orm macro where I can define the alias as well?

**Comment by tyt2y3** [comment]
Actually, can you try `let model: Model = serde_json::from_value(json);` ?

You can then convert it by `into_active_model`.

**Reply by cbishopvelti** [reply]
Not ideal as you get ActiveModel { id: Unchanged(0), a_b: Unchanged("ab") }

**Comment by tyt2y3** [comment]
Actually, @billy1624 why is `Transform attribute that exists in JSON object into ActiveValue::Set, otherwise ActiveValue::NotSet` needed? What problem does it solve?

**Reply by billy1624** [reply]
A field is missing from the JSON object, then, how do we represent it in the ActiveModel? We can only represent it with `ActiveValue::NotSet`.
